### ISSUE TITLE
`dotnet restore` fails when double-quotes used in PATH env var

### ISSUE THREAD

**Issue by natemcmaster** [dotnet/cli#827]
On Windows 10, using 1.0.0.1452673922

To repro: run `init-tools` from corefx.

```
Running: C:\Users\namc\dev\corefx\Tools\dotnetcli\bin\dotnet.exe restore C:\Users\namc\dev\corefx\Tools\1.0.25-prerelease-00148\project.json --packages C:\Users\namc\dev\corefx\packages\ --source https://www.myget.org/F/dotnet-buildtools/
Illegal characters in path.
Parameter name: path
```

**Comment by natemcmaster** [comment]
Found the offender: `%PATH%` included a double-quotes enclosed path, `...;"C:\Program Files\dotnet\bin";...` Removing the double quotes fixes the crash.

**Comment by piotrpMSFT** [comment]
Thanks @natemcmaster looks like a bug in our CommandResolver. This should affect every command, though, not just restore. Did you find that to be the case?

**Comment by natemcmaster** [comment]
I didn’t try anything other than restore…hard to compile w/o dependencies

**Comment by piotrpMSFT** [comment]
I thought there may have been a 'new' in there somewhere. Thanks.

**Comment by piotrpMSFT** [comment]
fixed by #863
